header/store/store.go

```diff
@@ -63,45 +50,58 @@ type store struct {
 	writeHead atomic.Pointer[header.ExtendedHeader]
 	// pending keeps headers pending to be written in one batch
 	pending *batch
+
+	params *Parameters
 }
 
 // NewStore constructs a Store over datastore.
 // The datastore must have a head there otherwise Start will error.
 // For first initialization of Store use NewStoreWithHead.
-func NewStore(ds datastore.Batching) (header.Store, error) {
-	return newStore(ds)
+func NewStore(ds datastore.Batching, opts ...Option) (header.Store, error) {
+	return newStore(ds, opts...)
 }
 
 // NewStoreWithHead initiates a new Store and forcefully sets a given trusted header as head.
-func NewStoreWithHead(ctx context.Context, ds datastore.Batching, head *header.ExtendedHeader) (header.Store, error) {
-	store, err := newStore(ds)
+func NewStoreWithHead(
+	ctx context.Context,
+	ds datastore.Batching,
+	head *header.ExtendedHeader,
+	opts ...Option,
+) (header.Store, error) {
+	store, err := newStore(ds, opts...)
 	if err != nil {
 		return nil, err
 	}
 
 	return store, store.Init(ctx, head)
 }
 
-func newStore(ds datastore.Batching) (*store, error) {
-	ds = namespace.Wrap(ds, storePrefix)
-	cache, err := lru.NewARC(DefaultStoreCacheSize)
+func newStore(ds datastore.Batching, opts ...Option) (*store, error) {
+	params := DefaultParameters()
+	for _, opt := range opts {
+		opt(params)
+	}
+
```

Comment:
Just realized that we also should do validation pkg level validation here.

Reply:
👍🏻